crates/compiler/src/lib.rs

```diff
@@ -15,19 +15,19 @@ pub fn compile_to_casm(
     let definition = serde_json::from_slice::<FeederGatewayContractClass<'_>>(sierra_definition)
         .context("Parsing Sierra class")?;
 
-    const V_0_11_0: semver::Version = semver::Version::new(0, 11, 0);
-    const V_0_11_1: semver::Version = semver::Version::new(0, 11, 1);
-    const V_0_11_2: semver::Version = semver::Version::new(0, 11, 2);
+    const V_0_11_0: StarknetVersion = StarknetVersion::new(0, 11, 0, 0);
+    const V_0_11_1: StarknetVersion = StarknetVersion::new(0, 11, 1, 0);
+    const V_0_11_2: StarknetVersion = StarknetVersion::new(0, 11, 2, 0);
 
     let result = std::panic::catch_unwind(|| {
-        match version
-            .parse_as_semver()
-            .context("Deciding on compiler version")?
-        {
-            Some(v) if v > V_0_11_2 => v2::compile(definition),
-            Some(v) if v > V_0_11_1 => v1_1_1::compile(definition),
-            Some(v) if v > V_0_11_0 => v1_0_0_rc0::compile(definition),
-            _ => v1_0_0_alpha6::compile(definition),
+        if version > &V_0_11_2 {
+            v2::compile(definition)
+        } else if version > &V_0_11_1 {
+            v1_1_1::compile(definition)
+        } else if version > &V_0_11_0 {
+            v1_0_0_rc0::compile(definition)
+        } else {
+            v1_0_0_alpha6::compile(definition)
```

Comment:
Minor nit: I found the match + if guards easier to read just because it has single lines.

As an aside, its a pity we can't use range matching on custom structs like this.

An "interesting" idea could also be to cast the version to u32, which would enable:

```rust
    const V_0_11_0: u32 = StarknetVersion::new(0, 11, 0, 0).into();
    const V_0_11_1: u32 = StarknetVersion::new(0, 11, 1, 0).into();
    const V_0_11_2: u32 = StarknetVersion::new(0, 11, 2, 0).into();

    let version: u32 = version.into();

    match version {
        ..V_0_11_0 => v1_0_0_alpha6::compile(definition),
        ..V_0_11_1 => v1_0_0_rc0::compile(definition),
        ..V_0_11_2 => v1_1_1_rc0::compile(definition),
        _          => v2::compile(definition)
    }
```
but definitely not worth it just for this 😁